spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/okhttp/OkHttpLoadBalancingClient.java

```diff
@@ -66,21 +78,36 @@ protected OkHttpClient createDelegate(IClientConfig config) {
 	}
 
 	@Override
-	public OkHttpRibbonResponse execute(OkHttpRibbonRequest ribbonRequest,
+	public OkHttpRibbonResponse execute(final OkHttpRibbonRequest ribbonRequest,
 			final IClientConfig configOverride) throws Exception {
-		boolean secure = isSecure(configOverride);
-
-		if (secure) {
-			final URI secureUri = UriComponentsBuilder.fromUri(ribbonRequest.getUri())
-					.scheme("https").build().toUri();
-			ribbonRequest = ribbonRequest.withNewUri(secureUri);
-		}
-
-		OkHttpClient httpClient = getOkHttpClient(configOverride, secure);
-
-		final Request request = ribbonRequest.toRequest();
-		Response response = httpClient.newCall(request).execute();
-		return new OkHttpRibbonResponse(response, ribbonRequest.getUri());
+		return this.executeWithRetry(ribbonRequest, new RetryCallback() {
+			@Override
+			public OkHttpRibbonResponse doWithRetry(RetryContext context) throws Exception {
+				//on retries the policy will choose the server and set it in the context
+				//extract the server and update the request being made
+				OkHttpRibbonRequest newRequest = ribbonRequest;
+				if(context instanceof LoadBalancedRetryContext) {
+					ServiceInstance service = ((LoadBalancedRetryContext)context).getServiceInstance();
+					if(service != null) {
+						//Reconstruct the request URI using the host and port set in the retry context
```

Comment:
It happens here https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonLoadBalancedRetryPolicyFactory.java#L73